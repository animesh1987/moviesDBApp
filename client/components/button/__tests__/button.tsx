import * as React from 'react';
import { create } from 'react-test-renderer';
import { shallow } from 'enzyme';

import { Button } from '../index';

describe('Movie Banner Test', () => {
  const isActive =  true;
  const isPrimary =  true;
  const isLong =  true;

  let clickFn = () => {};

  it('renders correctly', () => {
    const component = create(<Button onClick={clickFn()}/>)
      .toJSON();
    expect(component).toMatchSnapshot();
  });

  it('active added successfully', () => {
    const component = shallow(<Button isActive={isActive} />);
    expect(component.find('button').hasClass('active')).toBe(true);
  });

  it('active added successfully', () => {
    const component = shallow(<Button isLong={isLong} />);
    expect(component.find('button').hasClass('long')).toBe(true);
  });

  it('active added successfully', () => {
    const component = shallow(<Button isPrimary={isPrimary} />);
    expect(component.find('button').hasClass('primary')).toBe(true);
  });
});