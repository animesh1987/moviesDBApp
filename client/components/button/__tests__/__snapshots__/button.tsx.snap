// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Banner Test renders correctly 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;
